src/tests/psyGen_test.py

```diff
@@ -810,3 +815,843 @@ def test_invalid_reprod_pad_size():
+    do, omp directive and directive nodes'''
+    _, info = parse(os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                                 "test_files", "dynamo0p3",
+                                 "1_single_invoke.f90"),
+                    api="dynamo0.3")
+    psy = PSyFactory("dynamo0.3", distributed_memory=False).create(info)
+    invoke = psy.invokes.get('invoke_0_testkern_type')
+    schedule = invoke.schedule
+    from transformations import Dynamo0p3OMPLoopTrans, OMPParallelTrans
+    olooptrans = Dynamo0p3OMPLoopTrans()
+    ptrans = OMPParallelTrans()
+    # Put an OMP PARALLEL around this loop
+    child = schedule.children[0]
+    oschedule, _ = ptrans.apply(child)
+    # Put an OMP DO around this loop
+    schedule, _ = olooptrans.apply(oschedule.children[0].children[0])
+    # Replace the original loop schedule with the transformed one
+    omp_par_node = schedule.children[0]
+    assert omp_par_node.dag_name == "OMP_parallel_1"
+    assert omp_par_node.children[0].dag_name == "OMP_do_2"
+    omp_directive = super(OMPParallelDirective, omp_par_node)
+    assert omp_directive.dag_name == "OMP_directive_1"
+    print type(omp_directive)
+    directive = super(OMPDirective, omp_par_node)
+    assert directive.dag_name == "directive_1"
+
+EXPECTED = (
+    "digraph {\n"
+    "	schedule_start\n"
+    "	schedule_end\n"
+    "	loop_0_start\n"
+    "	loop_0_end\n"
+    "		loop_0_end -> schedule_end [color=blue]\n"
+    "		schedule_start -> loop_0_start [color=blue]\n"
+    "	kernel_testkern_code_2\n"
+    "		kernel_testkern_code_2 -> loop_0_end [color=blue]\n"
+    "		loop_0_start -> kernel_testkern_code_2 [color=blue]\n"
+    "}")
+
+
+def test_node_dag_no_graphviz(tmpdir):
+    '''test that dag generation does nothing if graphviz is not
+    installed. If graphviz is installed on this system we need to make
+    the test think that it is not. We do this by messing with
+    sys.modules. '''
+    import sys
+    keep = None
+    try:
+        import graphviz
+        keep = sys.modules['graphviz']
+        sys.modules['graphviz'] = None
+    except ImportError:
+        pass
+    _, invoke_info = parse(
+        os.path.join(BASE_PATH, "1_single_invoke.f90"),
+        distributed_memory=False, api="dynamo0.3")
+    psy = PSyFactory("dynamo0.3",
+                     distributed_memory=False).create(invoke_info)
+    invoke = psy.invokes.invoke_list[0]
+    schedule = invoke.schedule
+    my_file = tmpdir.join('test')
+    schedule.dag(file_name=my_file.strpath)
+    assert not os.path.exists(my_file.strpath)
+    if keep:
+        sys.modules['graphviz'] = keep
+
+
+def test_node_dag(tmpdir):
+    '''test that dag generation works correctly. Skip the test if
+    graphviz is not installed'''
+    graphviz_installed = False
+    try:
+        import graphviz
+        graphviz_installed = True
+    except ImportError:
+        pass
+    if graphviz_installed:
```

Comment:
Could you just return here (1633) and then you wouldn't have to have the rest of the test inside an `if` block?